plugins/lookup/merge_variables.py

```diff
@@ -0,0 +1,194 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2020, Thales Netherlands
+# Copyright (c) 2021, Ansible Project
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = """
+    author:
+      - Roy Lenferink (@rlenferink)
+      - Mark Ettema (@m-a-r-k-e)
+    name: merge_variables
+    short_description: merge variables with a certain suffix
+    description:
+        - This lookup returns the merged result of all variables in scope that matches the given suffix optionally
+          starting with an initial value.
+    version_added: 6.1.0
+    options:
+      _terms:
+        description:
+          - The suffix of the variable name (all available variables will be checked with endswith(suffix))
+        required: true
+      pattern_type:
+        description:
+          - Change the way of searching for the specified pattern.
+          - Possible values are C(prefix), C(suffix) and C(regex). By default C(suffix) is used.
+        type: str
+        default: 'suffix'
```

Comment:
How about using `regex` as the default? I think this is the most versatile and this would make the plugin behave similar to the `varnames` plugin by default.

Reply:
Initially I was thinking; this should be kept simple. But after learning that the behavior can be overwritten for every lookup call by setting the `ANSIBLE_MERGE_VARIABLES_PATTERN_TYPE` environment variable, I am okay with changing the default to at least have it consistent with the `varnames` plugin. :+1:

Reply:
Updated it!